bin/opam.ml

```diff
@@ -197,27 +197,25 @@ let submit ?distrib_uri ~token ~dry_run ~yes ~opam_repo ~user local_repo
   list_map Pkg.name pkgs >>= fun names ->
   let title = strf "[new release] %a (%s)" (pp_list Fmt.string) names version in
   Pkg.publish_msg pkg >>= fun changes ->
-  (match distrib_uri with Some uri -> Ok uri | None -> Pkg.infer_repo_uri pkg)
-  >>= Github_uri.get_user_and_repo
-  >>= fun (distrib_user, repo) ->
+  Pkg.infer_github_repo_uri pkg >>= fun { owner; repo } ->
```

Comment:
If I'm not missing something, this is a tiny breaking change: someone runs
```
dune-release opam submit --dist-uri <some_github_uri>
```
Before, that used to work. Now, it only works, if the distribution uri can be inferred from the opam file. Is that right? Do you think that can be ignored?

Reply:
I think this is very very unlikely and that this is inherently broken. To clarify, for that to happen we would need a user whose repo is not hosted on github that happened to have uploaded the distrib tarball somewhere on github. He passes that URI which we then use to determine a user and repo (which won't correspond to the one where the project is hosted, since it's not on github). We then either use them to rewrite the publication message in a github specific way and might also use that to open a PR to opam-repo if they have a broken configuration.

My impression is that would never work because it's using the wrong information in the wrong place.

The only case where it might eventually work is if the project is mirrored on github but even then, I'm not sure it would do the right thing. If we want to support mirroring on github (which we probably should) we'd need an CLI option so one can have the original repo in their opam file but pass the github mirror URI to dune-release so it release to the mirror as it would for a regular github hosted project.

To be perfectly honest I'm not 100% sure myself that I considered all the options but I feel like this has never worked for anyone. Do you have an example in mind of a situation where one would want this behaviour?